nemo/collections/asr/modules/transformer/transformer_generators.py

```diff
@@ -198,16 +241,29 @@ def _forward(
             pad_profile = torch.max(pad_profile, (next_tokens == self.eos).long())
             tgt = torch.cat((tgt, next_tokens.unsqueeze(1)), dim=-1)
 
+            if self.preserve_step_confidence:
+                if i == 0:
+                    step_confidence[0] = step_confidence[0].to(dtype=logits.dtype)
```

Comment:
Set the right dtype at initialization (encoder_hidden_states or decoder_input_ids should already have the right dtype)

Reply:
@pzelasko thanks, fixed!